File: pRobot169/python/dates.py
import time
import datetime

def CSTtime():
    return time.time() + 8 * 3600

def year_mon_day(t=None):
    if t is None:
        t = time.time()
    dt = datetime.datetime.fromtimestamp(t)
    return f"{dt.year} {dt.month:02d} {dt.day:02d}"

today = year_mon_day(CSTtime())
yesterday = year_mon_day(CSTtime() - 86400)
P2D = year_mon_day(CSTtime() - 2 * 86400)

print(" ".join([P2D, yesterday, today]))

File: pRobot169/perl/dates.pl
#!/usr/bin/perl
#

sub CSTtime () {
    return time+8*3600;
}

sub year_mon_day ($) {
    my $time = shift || time;
    my ($mday,$mon,$year_1900) = (gmtime($time))[3,4,5];
    my $year=$year_1900+1900;
    return sprintf("%04d %02d %02d", $year, $mon+1, $mday);
}

my $today = &year_mon_day(&CSTtime);
my $yesterday = &year_mon_day(&CSTtime - 86400);
my $P2D = &year_mon_day(&CSTtime - 2*86400);

print join(" ", $P2D, $yesterday, $today);